### PR TITLE
updating Addresses section

### PR DESCRIPTION
**Pull request by leerspace**
config.md: updating the Addresses section

Adding documentation on Announce and NoAnnounce.

Fixing typo in API default value.

License: MIT
Signed-off-by: John Reed <john@re2d.xyz>

### CODE REVIEW

**Comment by leerspace** [comment]
Ref: #3613

Closes #4325

File: docs/config.md [hunk 1]
```diff
@@ -42,6 +42,16 @@ Default:
 ]
 ```
 
+- `Announce`
+Array of swarm addresses to announce to the network.
+
+Default: `[]`
```
Comment (Stebalien) [inline docs/config.md]: This is a bit misleading. We announce all public addresses by default.

Comment (leerspace) [reply docs/config.md]: Would it be better to describe this as "Array of swarm addresses to announce to the network in addition to the ones that are inferred by default."?

Comment (lgierth) [reply docs/config.md]: If this is non-empty, we'll only announce what's in there. If it's empty, we'll announce the addresses that were inferred.

Comment (leerspace) [reply docs/config.md]: My bad. I was mixing up the "Swarm listening" lines and the "Swarm announcing" lines of the daemon output into a single bucket.

**Comment by whyrusleeping** [comment]
I wonder if gitcop just doesnt like windows line endings?

**Comment by leerspace** [comment]
Maybe. That or it just doesn't like me :).

FWIW this PR was made entirely using GitHub's built in forking/branching/editor tools, so that might also be a factor.

**Comment by whyrusleeping** [comment]
ah, interesting... oh well.

**Pull request merged**